test/test_um2netcdf.py

```diff
@@ -651,6 +663,355 @@ def test_fix_units_do_nothing_no_um_units(ua_plev_cube):
+            dummy_cube=ua_plev_cube,
+            coords=[lat_coordinate, lon_coordinate]
+        )
+
+        # Checks prior to modifications.
+        assert_coordinates_are_unmodified(lat_coordinate, lon_coordinate)
+
+        um2nc.fix_latlon_coords(cube_with_uv_coords, grid_type,
+                                D_LAT_N96, D_LON_N96)
+
+        assert lat_coordinate.var_name == um2nc.VAR_NAME_LAT_V
+        assert lon_coordinate.var_name == um2nc.VAR_NAME_LON_U
+
+        assert_dtype_float64(lat_coordinate, lon_coordinate)
+        assert_has_bounds(lat_coordinate, lon_coordinate)
+
+
+def test_fix_latlon_coords_standard(ua_plev_cube,
+                                    lat_standard_nd_coord,
+                                    lon_standard_nd_coord,
+                                    lat_standard_eg_coord,
+                                    lon_standard_eg_coord):
+    """
+    Tests of the fix_lat_lon_coords for standard longitude
+    and latitude coordinates on both the New Dynamics and
+    End Game grids.
+    """
+    coord_sets = [
+        (
+            lat_standard_nd_coord,
+            lon_standard_nd_coord,
+            um2nc.GRID_NEW_DYNAMICS
+         ),
+        (
+            lat_standard_eg_coord,
+            lon_standard_eg_coord,
+            um2nc.GRID_END_GAME
+        )
+    ]
+
+    for lat_coordinate, lon_coordinate, grid_type in coord_sets:
+
+        cube_with_uv_coords = DummyCubeWithCoords(
+            dummy_cube=ua_plev_cube,
+            coords=[lat_coordinate, lon_coordinate]
+        )
+
+        # Checks prior to modifications.
+        assert_coordinates_are_unmodified(lat_coordinate, lon_coordinate)
+
+        um2nc.fix_latlon_coords(cube_with_uv_coords, grid_type,
+                                D_LAT_N96, D_LON_N96)
+
+        assert lat_coordinate.var_name == um2nc.VAR_NAME_LAT_STANDARD
+        assert lon_coordinate.var_name == um2nc.VAR_NAME_LON_STANDARD
+
+        assert_dtype_float64(lat_coordinate, lon_coordinate)
+        assert_has_bounds(lat_coordinate, lon_coordinate)
+
+
+def test_fix_latlon_coords_single_point(ua_plev_cube):
+    """
+    Test that single point longitude and latitude coordinates
+    are provided with global bounds.
+    """
+
+    # Expected values after modification
+    expected_lat_bounds = um2nc.GLOBAL_COORD_BOUNDS[um2nc.LATITUDE]
+    expected_lon_bounds = um2nc.GLOBAL_COORD_BOUNDS[um2nc.LONGITUDE]
+
+    lat_coord_single = iris.coords.DimCoord(points=np.array([0]),
+                                            standard_name=um2nc.LATITUDE)
+    lon_coord_single = iris.coords.DimCoord(points=np.array([0]),
+                                            standard_name=um2nc.LONGITUDE)
+
+    cube_with_uv_coords = DummyCubeWithCoords(
+        dummy_cube=ua_plev_cube,
+        coords=[lat_coord_single, lon_coord_single]
+    )
+
+    assert not lat_coord_single.has_bounds()
+    assert not lon_coord_single.has_bounds()
```

Comment:
If the has bounds check helper is converted to a func returning a `bool`, this assertion can make use of it too.

Reply:
Yeah this works! Added in 6a4c876